tests/unit/test_fvm_multi.cpp

```diff
@@ -328,6 +329,337 @@ TEST(fvm_multi, mechanism_indexes)
     }
 }
 
+namespace {
+    double wm_impl(double wa, double xa) {
+        return wa? xa/wa: 0;
+    }
+
+    template <typename... R>
+    double wm_impl(double wa, double xa, double w, double x, R... rest) {
+        return wm_impl(wa+w, xa+w*x, rest...);
+    }
+
+    template <typename... R>
+    double wmean(double w, double x, R... rest) {
+        return wm_impl(w, w*x, rest...);
+    }
+}
+
+// Test area-weighted linear combination of density mechanism parameters.
+
+TEST(fvm_multi, density_weights) {
```

Comment:
Nice test, but you are right, it is a bit fragile.
I suppose it is the level you have to test at for this sort of thing.